core/src/main/java/io/micronaut/core/annotation/AnnotationMetadata.java

```diff
@@ -1459,9 +1464,9 @@ default boolean hasDeclaredAnnotation(@Nullable Class<? extends Annotation> anno
      */
     default boolean hasDeclaredStereotype(@Nullable Class<? extends Annotation> stereotype) {
         if (stereotype != null) {
-            Repeatable repeatable = stereotype.getAnnotation(Repeatable.class);
-            if (repeatable != null) {
-                return hasDeclaredStereotype(repeatable.value().getName());
+            Optional<String> repeatable = findRepeatableAnnotation(stereotype);
```

Comment:
For these internal calls maybe we can avoid optional wrapping?

Reply:
It's a bit problematic as it's an interface and we cannot have private methods. I can call `isRepeatableAnnotation` before getting it considering in most cases it's probably not repeatable. WDYT?

Reply:
Sounds good